src/haystack/filter/parser.rs

```diff
@@ -397,6 +397,24 @@ mod test {
         assert_eq!(parens.to_string(), "a or ( b and c )");
     }
 
+    #[test]
+    fn test_filter_parser_siteref_site_and_equip() {
+        let mut input = Cursor::new("siteRef->dis and equip".as_bytes());
+        let mut parser = Parser::make(&mut input).expect("Should create parser");
+
+        let parens = parser.parse().expect("Parens");

```

Comment:
Some copy pasta here for the expect. Should be fixed to show the correct message. There are some more like this bellow